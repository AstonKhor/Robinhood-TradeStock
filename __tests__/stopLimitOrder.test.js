/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import StopLimitOrder from '../client/components/stopLimitOrder.jsx';

describe('<StopLimitOrder /> Component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<StopLimitOrder />);
    expect(tree).toMatchSnapshot();
  });

  it('initiates state as expected', () => {
    const wrapper = shallow(<StopLimitOrder />);
    expect(wrapper.state().stopPrice).toBe(0);
    expect(wrapper.state().shares).toBe(0);
    expect(wrapper.state().expires).toBe('Good for Day');
  });
});
