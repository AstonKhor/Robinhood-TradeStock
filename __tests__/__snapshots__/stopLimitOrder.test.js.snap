// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StopLimitOrder /> Component matches the snapshot 1`] = `
Array [
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Stop Price
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="stopPrice"
      onChange={[Function]}
      placeholder="$0.00"
      step=".01"
      type="number"
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Limit Price
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="limitPrice"
      onChange={[Function]}
      placeholder="$0.00"
      step=".01"
      type="number"
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Shares
    </label>
    <input
      className="sc-bdVaJa sc-htpNat dFFWgj"
      name="shares"
      onChange={[Function]}
      placeholder="0"
      type="number"
    />
  </div>,
  <div
    className="sc-bxivhb cyWHkv"
  >
    <label
      className="sc-bwzfXH dbwzZe"
    >
      Expires
    </label>
    <div
      className="sc-iwsKbI bbqNLX"
      onClick={[Function]}
    >
      Good for Day
    </div>
  </div>,
]
`;
